transaction-view/src/transaction_meta.rs

```diff
@@ -0,0 +1,393 @@
+    /// The `bytes` parameter must have no trailing data.
+    pub fn try_new(bytes: &[u8]) -> Result<Self> {
+        let mut offset = 0;
+        let signature = SignatureMeta::try_new(bytes, &mut offset)?;
+        let message_header = MessageHeaderMeta::try_new(bytes, &mut offset)?;
+        let static_account_keys = StaticAccountKeysMeta::try_new(bytes, &mut offset)?;
+
+        // The recent blockhash is the first account key after the static
+        // account keys. The recent blockhash is always present in a valid
+        // transaction and has a fixed size of 32 bytes.
+        let recent_blockhash_offset = offset as u16;
+        advance_offset_for_type::<Hash>(bytes, &mut offset)?;
+
+        let instructions = InstructionsMeta::try_new(bytes, &mut offset)?;
+        let address_table_lookup = match message_header.version {
+            TransactionVersion::Legacy => AddressTableLookupMeta {
+                num_address_table_lookup: 0,
+                offset: 0,
+            },
+            TransactionVersion::V0 => AddressTableLookupMeta::try_new(bytes, &mut offset)?,
+        };
+
+        // Verify that the entire transaction was parsed.
+        if offset != bytes.len() {
+            return Err(TransactionParsingError);
+        }
+
+        Ok(Self {
+            signature,
+            message_header,
+            static_account_keys,
+            recent_blockhash_offset,
+            instructions,
+            address_table_lookup,
+        })
+    }
+
+    /// Return the number of signatures in the transaction.
+    pub fn num_signatures(&self) -> u8 {
+        self.signature.num_signatures
+    }
+
+    /// Return the version of the transaction.
+    pub fn version(&self) -> TransactionVersion {
+        self.message_header.version
+    }
+
+    /// Return the number of required signatures in the transaction.
+    pub fn num_required_signatures(&self) -> u8 {
+        self.message_header.num_required_signatures
+    }
+
+    /// Return the number of readonly signed accounts in the transaction.
+    pub fn num_readonly_signed_accounts(&self) -> u8 {
+        self.message_header.num_readonly_signed_accounts
+    }
+
+    /// Return the number of readonly unsigned accounts in the transaction.
+    pub fn num_readonly_unsigned_accounts(&self) -> u8 {
+        self.message_header.num_readonly_unsigned_accounts
+    }
+
+    /// Return the number of static account keys in the transaction.
+    pub fn num_static_account_keys(&self) -> u8 {
+        self.static_account_keys.num_static_accounts
+    }
+
+    /// Return the number of instructions in the transaction.
+    pub fn num_instructions(&self) -> u16 {
+        self.instructions.num_instructions
+    }
+
+    /// Return the number of address table lookups in the transaction.
+    pub fn num_address_table_lookups(&self) -> u8 {
+        self.address_table_lookup.num_address_table_lookup
+    }
+}
+
+// Separate implementation for `unsafe` accessor methods.
```

Comment:
While these are `unsafe` on the non-owning `TransactionMeta`, they will be safe on the final struct that owns the data + this meta.

As mentioned in the description, the meta is separate and non-owning so we can use this in places where we do not want to own the data.